feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/vote/validations/VoteReferendumValidationPayload.kt

```diff
@@ -1,15 +1,19 @@
 package io.novafoundation.nova.feature_governance_impl.domain.referendum.vote.validations
 
 import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.OnChainReferendum
+import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.VoteType
 import io.novafoundation.nova.feature_governance_api.data.network.blockhain.model.Voting
 import io.novafoundation.nova.feature_wallet_api.domain.model.Asset
 import io.novafoundation.nova.feature_wallet_api.presentation.model.DecimalFee
+import io.novafoundation.nova.runtime.multiNetwork.runtime.types.custom.vote.Conviction
 import java.math.BigDecimal
 
-class VoteReferendumValidationPayload(
+data class VoteReferendumValidationPayload(
     val onChainReferendum: OnChainReferendum,
     val asset: Asset,
     val trackVoting: Voting?,
     val voteAmount: BigDecimal,
-    val fee: DecimalFee
+    val fee: DecimalFee,
+    val voteType: VoteType,
+    val conviction: Conviction
```

Comment:
As conviction for abstain doesn't make sense I would suggest to have the conviction as associated value of the VoteType just to not having mismatching between voteType and conviction

Reply:
Since this is validation payload it's important to pass conviction here to check that Conviction is NONE for abstain vote and show a dialog.
But this is a valid point for confirmation screen payload. Thank you for this advice.